crates/optimism/trie/src/prune/task.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{prune::OpProofStoragePruner, OpProofsStore};
+use reth_provider::BlockHashReader;
+use tokio::{time, time::Duration};
+use tokio_util::sync::CancellationToken;
+use tracing::info;
+
+/// Periodic pruner task: constructs the pruner and runs it every interval.
+#[derive(Debug)]
+pub struct OpProofStoragePrunerTask<P, H> {
+    pruner: OpProofStoragePruner<P, H>,
+    min_block_interval: u64,
+    task_run_interval: Duration,
+    cancel: CancellationToken,
+}
+
+impl<P, H> OpProofStoragePrunerTask<P, H>
+where
+    P: OpProofsStore,
+    H: BlockHashReader,
+{
+    /// Initialize a new ` OpProofStoragePrunerTask `
```

Comment:
```suggestion
    /// Initialize a new [`OpProofStoragePrunerTask`]
```
nitpick, allows the compiler to help us catch any potential name change in the future